develop: archived -> legacy (#11990)
pingcap/docs: develop/dev-guide-outdated-for-django.md

> **Note:**
>

> > This document is a legacy document. This indicates that this document will not be updated thereafter. You can see [Developer Guide Overview](/develop/dev-guide-overview.md) for more details.

Reviewer: ```suggestion
> This legacy document is outdated and will not be updated thereafter. You can see [Developer Guide Overview](/develop/dev-guide-overview.md) for more details.
```
